sql/tests: TestRandomSyntaxGeneration failed

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/21356540?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/21356540?buildTab=artifacts#/) on master @ [248e519687c1d8129f5675a0fe0db9cf125c2b6d](https://github.com/cockroachdb/cockroach/commits/248e519687c1d8129f5675a0fe0db9cf125c2b6d):


```
    rsg_test.go:125: Last executed (4): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (5): DO LANGUAGE SECONDARY LANGUAGE 'string' LANGUAGE TIME 'string' 'string'
    rsg_test.go:125: Last executed (6): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (7): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (8): COMMENT ON DATABASE OPERATOR IS 'string'
    rsg_test.go:125: Last executed (9): DO LANGUAGE 'string' LANGUAGE GROUPING 'string'
    rsg_test.go:125: Last executed (10): LOCK ONLY RESOLVED . REGIONAL . FAMILY IN SHARE MODE
    rsg_test.go:125: Last executed (11): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (12): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (13): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (14): LOCK ident , ONLY ( INDEX_BEFORE_NAME_THEN_PAREN . FAMILY . ident ) , FAMILY . POLYGON . ident * , ROW . OUTER . CASE , ONLY DECIMAL , ONLY INDEX_BEFORE_NAME_THEN_PAREN , SKIP_MISSING_UDFS
    rsg_test.go:125: Last executed (15): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (16): COPY TIMESTAMP FROM STDIN WITH CSV
    rsg_test.go:125: Last executed (17): BACKUP TENANT 443 INTO PLACEHOLDER WITH OPTIONS ( INCLUDE_ALL_SECONDARY_TENANTS )
    rsg_test.go:125: Last executed (18): REASSIGN OWNED BY SESSION_USER , INDEX TO SKIP
    rsg_test.go:125: Last executed (19): LOCK TABLE ONLY INDEX_BEFORE_PAREN , FAMILY * IN ROW EXCLUSIVE MODE NOWAIT
    rsg_test.go:125: Last executed (20): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (21): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (22): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (23): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (24): PREPARE PER AS IMPORT INTO FAMILY . STRING ( DEFAULTS . error , REAL ) POINT DATA ( PLACEHOLDER )
    rsg_test.go:125: Last executed (25): COPY FAMILY . ILIKE ( DECIMAL ) TO STDOUT
    rsg_test.go:125: Last executed (26): ANALYSE INT . ILIKE . EXTRACT_DURATION
    rsg_test.go:125: Last executed (27): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (28): LOCK FAMILY . ident . CONSTRAINT * , ident *
    rsg_test.go:125: Last executed (29): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (30): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (31): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (32): FETCH RELATIVE - -319 FROM SUBSTRING
    rsg_test.go:125: Last executed (33): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (34): PREPARE ident ( POLYGON error ARRAY [ -78 ] [ error ) AS DROP VIRTUAL CLUSTER error
    rsg_test.go:125: Last executed (35): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (36): MOVE BACKWARD ALL IN RESTART
    rsg_test.go:125: Last executed (37): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (38): PREPARE ENUMS AS EXPLAIN SET LOCAL error
    rsg_test.go:125: Last executed (39): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (40): MOVE -9 GEOMETRY
    rsg_test.go:125: Last executed (41): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (42): FETCH RELATIVE 292 LABEL
    rsg_test.go:125: Last executed (43): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (44): TRUNCATE ONLY ( INDEX_BEFORE_PAREN ) , ONLY INDEX_BEFORE_PAREN , FAMILY * , FAMILY * , INDEX_AFTER_ORDER_BY_BEFORE_AT . CHECK * , ONLY ( INDEX_AFTER_ORDER_BY_BEFORE_AT . ident )
    rsg_test.go:125: Last executed (45): LOCK ONLY ( FAMILY ) , BOOLEAN , INDEX_AFTER_ORDER_BY_BEFORE_AT . CHARACTERISTICS , ONLY ( INDEX_AFTER_ORDER_BY_BEFORE_AT ) , ONLY ( ident ) IN ACCESS SHARE MODE NOWAIT
    rsg_test.go:125: Last executed (46): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (47): LOCK TABLE INDEX_BEFORE_NAME_THEN_PAREN . LEFT , ONLY ( NOTHING_AFTER_RETURNING . UNIQUE . UNSPLIT ) IN SHARE MODE NOWAIT
    rsg_test.go:125: Last executed (48): ANALYZE RETURNS . POINTZ
    rsg_test.go:125: Last executed (49): LOCK FAMILY . CURRENT_USER , PREPARE * NOWAIT
    rsg_test.go:128: Currently active statements: map[]
    rsg_test.go:377: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration1127125735
--- FAIL: TestRandomSyntaxGeneration (303.43s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-64173

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/21357923?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/21357923?buildTab=artifacts#/) on master @ [971ba39ae57f213d10f07b1cadad5404361fa66f](https://github.com/cockroachdb/cockroach/commits/971ba39ae57f213d10f07b1cadad5404361fa66f):


```
    rsg_test.go:125: Last executed (4): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (5): DROP OWNED BY ident , CURRENT_USER , FORCE_QUOTE , VALIDATE , ident CASCADE
    rsg_test.go:125: Last executed (6): DECLARE LEAST BINARY CURSOR WITHOUT HOLD FOR_JOB ( error ORDER BY PRIMARY KEY FAMILY ASC FOR NO KEY UPDATE
    rsg_test.go:125: Last executed (7): UNLISTEN ident . ident . FLOAT
    rsg_test.go:125: Last executed (8): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (9): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (10): COPY POSITION . ident . ident ( KEYS ) FROM error
    rsg_test.go:125: Last executed (11): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (12): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (13): LOCK ONLY ( ident . NOCREATEDB . CURRENT_TIMESTAMP ) NOWAIT
    rsg_test.go:125: Last executed (14): PREPARE FORCE_INVERTED_INDEX ( ident ARRAY ) AS OPT PLAN 'string'
    rsg_test.go:125: Last executed (15): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (16): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (17): MOVE FORWARD FROM ROW
    rsg_test.go:125: Last executed (18): UNLISTEN FAMILY . AFTER . DECIMAL
    rsg_test.go:125: Last executed (19): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (20): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (21): LOCK TABLE FAMILY . PRECISION . CURSOR IN EXCLUSIVE MODE NOWAIT
    rsg_test.go:125: Last executed (22): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (23): PREPARE ident AS IMPORT INTO ident ident DATA ( PLACEHOLDER , 'string' ) WITH OPTIONS ( 'string' , VALUE = PLACEHOLDER )
    rsg_test.go:125: Last executed (24): PREPARE VARBIT ( LIKE . COLLATION ARRAY , POINT error ARRAY [ -401 ] ) AS IMPORT error
    rsg_test.go:125: Last executed (25): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (26): DO LANGUAGE TEXT 'string'
    rsg_test.go:125: Last executed (27): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (28): COMMENT ON TABLE FORCE_NOT_NULL . VARBIT IS 'string'
    rsg_test.go:125: Last executed (29): DROP OWNED BY SESSION_USER , REFERENCING , ident
    rsg_test.go:125: Last executed (30): COMMENT ON TYPE FAMILY . ident . POINTZM IS 'string'
    rsg_test.go:125: Last executed (31): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (32): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (33): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (34): MOVE LAST FROM LAST
    rsg_test.go:125: Last executed (35): DROP OWNED BY SCROLL , ident , VARIABLES , LABEL , SESSION_USER , CURRENT_USER
    rsg_test.go:125: Last executed (36): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (37): LOCK ONLY CHAR , ONLY INDEX_BEFORE_PAREN . ident . ident , ONLY ( ROW . ADMIN ) , FAMILY . ident NOWAIT
    rsg_test.go:125: Last executed (38): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (39): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (40): FETCH - -285 FROM ANNOTATE_TYPE
    rsg_test.go:125: Last executed (41): PREPARE ident ( BIT VARYING ( 309 ) ARRAY , POLYGON error [ ] ) AS OPT PLAN 'string'
    rsg_test.go:125: Last executed (42): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (43): COPY INDEX_BEFORE_PAREN . ELSE . ROLLBACK ( OWNER ) FROM error
    rsg_test.go:125: Last executed (44): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (45): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (46): LOCK ONLY ( ident ) , ONLY ( FAMILY ) IN SHARE ROW EXCLUSIVE MODE NOWAIT
    rsg_test.go:125: Last executed (47): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (48): PAUSE ALL OVERLAY JOBS
    rsg_test.go:125: Last executed (49): PREPARE ident ( VARBIT ( -265 ) ARRAY ) AS EXPERIMENTAL SCRUB TABLE JOB . WINDOW
    rsg_test.go:128: Currently active statements: map[]
    rsg_test.go:377: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration3129934039
--- FAIL: TestRandomSyntaxGeneration (302.91s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/21358708?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/21358708?buildTab=artifacts#/) on master @ [b852382ca5d1b265a84370597f0bd77c0ab110cb](https://github.com/cockroachdb/cockroach/commits/b852382ca5d1b265a84370597f0bd77c0ab110cb):


```
    rsg_test.go:125: Last executed (4): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (5): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (6): LOCK TABLE FAMILY * , ONLY ( FAMILY ) , ONLY ( INDEX_BEFORE_NAME_THEN_PAREN . IS ) IN SHARE UPDATE EXCLUSIVE MODE
    rsg_test.go:125: Last executed (7): PREPARE GEOGRAPHY ( CHARACTER VARYING ARRAY [ -177 ] [ error ) AS OPT PLAN 'string'
    rsg_test.go:125: Last executed (8): REASSIGN OWNED BY DEALLOCATE TO DEFINER
    rsg_test.go:125: Last executed (9): COPY FAMILY . ONLY ( POSITION ) TO STDOUT WITH QUOTE 'string'
    rsg_test.go:125: Last executed (10): DROP OWNED BY ident , SESSION_USER , ident , CURRENT_USER , NEW
    rsg_test.go:125: Last executed (11): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (12): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (13): COPY GREATEST ( GENERATED , NODE , ident , ident ) FROM error
    rsg_test.go:125: Last executed (14): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (15): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (16): PREPARE COVERING ( @ 439 ARRAY [ 50 ] ) AS SHOW SURVIVAL GOAL FROM DATABASE PREPARE
    rsg_test.go:125: Last executed (17): COPY FAMILY . ANALYZE . FAMILY ( ident ) TO error
    rsg_test.go:125: Last executed (18): RESTORE STATE FROM error
    rsg_test.go:125: Last executed (19): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (20): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (21): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (22): PREPARE CAPABILITY ( @ 203 ARRAY [ -402 ] , POLYGON error ARRAY [ 25 ] , INTERVAL ARRAY [ -279 ] [ error ) AS OPT PLAN 'string'
    rsg_test.go:125: Last executed (23): COPY ( ( WITH RECURSIVE SYSTEM ( ident , ident ) AS NOT MATERIALIZED ( ALTER EXTERNAL CONNECTION IF EXISTS IF NOT EXISTS PLACEHOLDER AS PLACEHOLDER ) TABLE error ) ) TO STDOUT WITH FORCE QUOTE error BINARY CSV
    rsg_test.go:125: Last executed (24): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (25): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (26): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (27): COPY FAMILY . LOCALTIMESTAMP . FAMILY FROM STDIN
    rsg_test.go:125: Last executed (28): PREPARE DETACHED ( VARBIT ARRAY [ 270 ] ) AS ( VALUES error )
    rsg_test.go:125: Last executed (29): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (30): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (31): ANALYZE EXPLAIN . LIMIT
    rsg_test.go:125: Last executed (32): DROP OWNED BY DEBUG , POINTZM RESTRICT
    rsg_test.go:125: Last executed (33): LOCK FAMILY . TIMESTAMP * , ONLY ( INDEX_AFTER_ORDER_BY_BEFORE_AT . RETURN ) NOWAIT
    rsg_test.go:125: Last executed (34): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (35): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (36): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (37): DROP OWNED BY ident , ident , SESSION_USER , ident , SESSION_USER , SESSION_USER RESTRICT
    rsg_test.go:125: Last executed (38): COPY ident . SYMMETRIC . KEYS TO error
    rsg_test.go:125: Last executed (39): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (40): DROP OWNED BY SESSION_USER , SETTINGS , CURRENT_USER , SESSION_USER , CURRENT_USER CASCADE
    rsg_test.go:125: Last executed (41): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (42): LOCK TABLE CONFIGURATIONS * , ONLY ( NOTHING_AFTER_RETURNING . OLD ) , FAMILY . WINDOW * , FLOAT IN ACCESS EXCLUSIVE MODE NOWAIT
    rsg_test.go:125: Last executed (43): REFRESH MATERIALIZED VIEW CONCURRENTLY NEW_KMS . FAMILY WITH NO DATA
    rsg_test.go:125: Last executed (44): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (45): CREATE DATABASE IF NOT EXISTS ident
    rsg_test.go:125: Last executed (46): LOCK TABLE ONLY FAMILY . GEOGRAPHY , FAMILY . TIMESTAMP . ident IN ROW EXCLUSIVE MODE
    rsg_test.go:125: Last executed (47): LOCK ONLY ( FAMILY ) , VIEWCLUSTERSETTING , ONLY ( NULLIF ) , ONLY ( COALESCE ) NOWAIT
    rsg_test.go:125: Last executed (48): CREATE SCHEDULE 'string' FOR BACKUP INTO ident WITH OPTIONS ( STRICT STORAGE LOCALITY ) RECURRING PLACEHOLDER
    rsg_test.go:125: Last executed (49): PREPARE VALUE ( POINT error ARRAY , FORCE_INDEX . ident . INCREMENT ARRAY [ 287 ] , POINT error ARRAY [ 241 ] ) AS CREATE RESOURCE GROUP IF NOT EXISTS PREPARED WITH ident = ON
    rsg_test.go:128: Currently active statements: map[]
    rsg_test.go:377: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration2473368205
--- FAIL: TestRandomSyntaxGeneration (303.32s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/docs/generated/sql/bnf/stmt_block.bnf
+++ b/docs/generated/sql/bnf/stmt_block.bnf
@@ -1946,10 +1946,7 @@ label_spec ::=
 	| 'IF' 'NOT' 'EXISTS' string_or_placeholder
 
 with_resource_group_options ::=
-	'WITH' resource_group_option_list
-	| 'WITH' resource_group_option_list
-	| 'WITH' '(' resource_group_option_list ')'
-	| 'WITH' '(' resource_group_option_list ')'
+	'WITH' '(' resource_group_option_list ')'
 
 role_or_group_or_user ::=
 	'ROLE'

--- a/pkg/sql/logictest/testdata/logic_test/resource_group
+++ b/pkg/sql/logictest/testdata/logic_test/resource_group
@@ -7,10 +7,10 @@
 subtest disabled_by_default
 
 statement error pq: resource group SQL is disabled
-CREATE RESOURCE GROUP g WITH cpu_weight = 100
+CREATE RESOURCE GROUP g WITH (cpu_weight = 100)
 
 statement error pq: resource group SQL is disabled
-ALTER RESOURCE GROUP g WITH cpu_weight = 100
+ALTER RESOURCE GROUP g WITH (cpu_weight = 100)
 
 statement error pq: resource group SQL is disabled
 DROP RESOURCE GROUP g
@@ -34,28 +34,32 @@ statement error at or near "EOF": syntax error
 CREATE RESOURCE GROUP missing_weight
 
 statement error pq: cpu_weight is required and must be a positive integer
-CREATE RESOURCE GROUP only_max_cpu WITH max_cpu = true
+CREATE RESOURCE GROUP only_max_cpu WITH (max_cpu = true)
 
 statement error pq: cpu_weight must be a positive integer
-CREATE RESOURCE GROUP zero_weight WITH cpu_weight = 0
+CREATE RESOURCE GROUP zero_weight WITH (cpu_weight = 0)
 
 statement error pq: cpu_weight must be a positive integer
-CREATE RESOURCE GROUP negative_weight WITH cpu_weight = -1
+CREATE RESOURCE GROUP negative_weight WITH (cpu_weight = -1)
 
 # Unknown options are rejected.
 statement error pq: unknown resource group option "bogus"
-CREATE RESOURCE GROUP unknown_opt WITH cpu_weight = 1, bogus = 1
+CREATE RESOURCE GROUP unknown_opt WITH (cpu_weight = 1, bogus = 1)
 
 # Repeating an option is rejected (rather than silently last-writer-wins).
 statement error pq: cpu_weight specified multiple times
-CREATE RESOURCE GROUP dup_weight WITH cpu_weight = 10, cpu_weight = 20
+CREATE RESOURCE GROUP dup_weight WITH (cpu_weight = 10, cpu_weight = 20)
 
 statement error pq: max_cpu specified multiple times
-CREATE RESOURCE GROUP dup_max_cpu WITH cpu_weight = 1, max_cpu = true, max_cpu = false
+CREATE RESOURCE GROUP dup_max_cpu WITH (cpu_weight = 1, max_cpu = true, max_cpu = false)
+
+# The no-parens form is rejected; parentheses are required around the option list.
+statement error at or near "cpu_weight": syntax error
+CREATE RESOURCE GROUP no_parens WITH cpu_weight = 100
 
 # IDs < 16 are reserved; the sequence's first allocated id is 17.
 statement ok
-CREATE RESOURCE GROUP high WITH cpu_weight = 100, max_cpu = true
+CREATE RESOURCE GROUP high WITH (cpu_weight = 100, max_cpu = true)
 
 query ITIBF colnames
 SHOW RESOURCE GROUP high
@@ -63,29 +67,22 @@ SHOW RESOURCE GROUP high
 id  name  cpu_weight  max_cpu  cpu_share_percent
 17  high  100         true     100
 
-# The parens form is also accepted.
-statement ok
-CREATE RESOURCE GROUP parens WITH (cpu_weight = 50, max_cpu = false)
-
-statement ok
-DROP RESOURCE GROUP parens
-
 # Defaults: max_cpu omitted on CREATE -> false.
 statement ok
-CREATE RESOURCE GROUP medium WITH cpu_weight = 50
+CREATE RESOURCE GROUP medium WITH (cpu_weight = 50)
 
 query ITIBF colnames
 SHOW RESOURCE GROUP medium
 ----
 id  name    cpu_weight  max_cpu  cpu_share_percent
-19  medium  50          false    33.33333333333333
+18  medium  50          false    33.33333333333333
 
 # Duplicate name is rejected; IF NOT EXISTS swallows the conflict.
 statement error pq: resource group "high" already exists
-CREATE RESOURCE GROUP high WITH cpu_weight = 999
+CREATE RESOURCE GROUP high WITH (cpu_weight = 999)
 
 statement ok
-CREATE RESOURCE GROUP IF NOT EXISTS high WITH cpu_weight = 999
+CREATE RESOURCE GROUP IF NOT EXISTS high WITH (cpu_weight = 999)
 
 # The existing row was not modified.
 query I
@@ -103,7 +100,7 @@ SHOW RESOURCE GROUPS
 ----
 id  name    cpu_weight  max_cpu  cpu_share_percent
 17  high    100         true     66.66666666666666
-19  medium  50          false    33.33333333333333
+18  medium  50          false    33.33333333333333
 
 # SHOW RESOURCE GROUP <name> errors when the named group does not exist.
 statement error pq: resource group "ghost" does not exist
@@ -115,7 +112,7 @@ subtest alter
 
 # ALTER updates only specified options (partial update).
 statement ok
-ALTER RESOURCE GROUP high WITH cpu_weight = 200
+ALTER RESOURCE GROUP high WITH (cpu_weight = 200)
 
 query ITIBF colnames
 SHOW RESOURCE GROUP high
@@ -124,7 +121,7 @@ id  name  cpu_weight  max_cpu  cpu_share_percent
 17  high  200         true     80
 
 statement ok
-ALTER RESOURCE GROUP high WITH max_cpu = false
+ALTER RESOURCE GROUP high WITH (max_cpu = false)
 
 query ITIBF colnames
 SHOW RESOURCE GROUP high
@@ -134,14 +131,14 @@ id  name  cpu_weight  max_cpu  cpu_share_percent
 
 # ALTER on a missing group: error without IF EXISTS, no-op with.
 statement error pq: resource group "ghost" does not exist
-ALTER RESOURCE GROUP ghost WITH cpu_weight = 1
+ALTER RESOURCE GROUP ghost WITH (cpu_weight = 1)
 
 statement ok
-ALTER RESOURCE GROUP IF EXISTS ghost WITH cpu_weight = 1
+ALTER RESOURCE GROUP IF EXISTS ghost WITH (cpu_weight = 1)
 
 # Repeating an option is rejected on ALTER as well.
 statement error pq: cpu_weight specified multiple times
-ALTER RESOURCE GROUP high WITH cpu_weight = 10, cpu_weight = 20
+ALTER RESOURCE GROUP high WITH (cpu_weight = 10, cpu_weight = 20)
 
 subtest end
 
@@ -179,13 +176,13 @@ SELECT cpu_share_percent FROM [SHOW RESOURCE GROUP high]
 100
 
 statement ok
-ALTER RESOURCE GROUP high WITH cpu_weight = 20
+ALTER RESOURCE GROUP high WITH (cpu_weight = 20)
 
 statement ok
-CREATE RESOURCE GROUP a WITH cpu_weight = 30
+CREATE RESOURCE GROUP a WITH (cpu_weight = 30)
 
 statement ok
-CREATE RESOURCE GROUP b WITH cpu_weight = 50
+CREATE RESOURCE GROUP b WITH (cpu_weight = 50)
 
 query TF rowsort
 SELECT name, cpu_share_percent FROM [SHOW RESOURCE GROUPS]
@@ -211,7 +208,7 @@ statement ok
 DROP RESOURCE GROUP a
 
 statement ok
-ALTER RESOURCE GROUP high WITH cpu_weight = 200
+ALTER RESOURCE GROUP high WITH (cpu_weight = 200)
 
 subtest end
 
@@ -220,10 +217,10 @@ subtest non_admin
 user testuser
 
 statement error pq: only users with the admin role may use resource group statements
-CREATE RESOURCE GROUP unauth WITH cpu_weight = 1
+CREATE RESOURCE GROUP unauth WITH (cpu_weight = 1)
 
 statement error pq: only users with the admin role may use resource group statements
-ALTER RESOURCE GROUP high WITH cpu_weight = 1
+ALTER RESOURCE GROUP high WITH (cpu_weight = 1)
 
 statement error pq: only users with the admin role may use resource group statements
 DROP RESOURCE GROUP high

--- a/pkg/sql/parser/sql.y
+++ b/pkg/sql/parser/sql.y
@@ -4285,19 +4285,7 @@ drop_external_connection_stmt:
 	| DROP EXTERNAL CONNECTION error // SHOW HELP: DROP EXTERNAL CONNECTION
 
 with_resource_group_options:
-  WITH resource_group_option_list
-  {
-    $$.val = $2.kvOptions()
-  }
-| WITH_LA resource_group_option_list
-  {
-    $$.val = $2.kvOptions()
-  }
-| WITH '(' resource_group_option_list ')'
-  {
-    $$.val = $3.kvOptions()
-  }
-| WITH_LA '(' resource_group_option_list ')'
+  WITH '(' resource_group_option_list ')'
   {
     $$.val = $3.kvOptions()
   }

--- a/pkg/sql/parser/testdata/resource_group
+++ b/pkg/sql/parser/testdata/resource_group
@@ -15,15 +15,6 @@ CREATE RESOURCE GROUP my_group WITH (cpu_weight = (100), max_cpu = (false)) -- f
 CREATE RESOURCE GROUP my_group WITH (cpu_weight = _, max_cpu = _) -- literals removed
 CREATE RESOURCE GROUP _ WITH (cpu_weight = 100, max_cpu = false) -- identifiers removed
 
-# No-parens form is also accepted; the canonical form adds them.
-parse
-CREATE RESOURCE GROUP my_group WITH cpu_weight = 100, max_cpu = false
-----
-CREATE RESOURCE GROUP my_group WITH (cpu_weight = 100, max_cpu = false) -- normalized!
-CREATE RESOURCE GROUP my_group WITH (cpu_weight = (100), max_cpu = (false)) -- fully parenthesized
-CREATE RESOURCE GROUP my_group WITH (cpu_weight = _, max_cpu = _) -- literals removed
-CREATE RESOURCE GROUP _ WITH (cpu_weight = 100, max_cpu = false) -- identifiers removed
-
 parse
 CREATE RESOURCE GROUP IF NOT EXISTS my_group WITH (cpu_weight = 1, max_cpu = true)
 ----
@@ -32,7 +23,6 @@ CREATE RESOURCE GROUP IF NOT EXISTS my_group WITH (cpu_weight = (1), max_cpu = (
 CREATE RESOURCE GROUP IF NOT EXISTS my_group WITH (cpu_weight = _, max_cpu = _) -- literals removed
 CREATE RESOURCE GROUP IF NOT EXISTS _ WITH (cpu_weight = 1, max_cpu = true) -- identifiers removed
 
-# Ensure BUCKET_COUNT (which triggers WITH_LA in the lexer) works correctly.
 parse
 CREATE RESOURCE GROUP my_group WITH (bucket_count = 10)
 ----
@@ -42,26 +32,9 @@ CREATE RESOURCE GROUP my_group WITH (bucket_count = _) -- literals removed
 CREATE RESOURCE GROUP _ WITH (bucket_count = 10) -- identifiers removed
 
 parse
-CREATE RESOURCE GROUP my_group WITH bucket_count = 10
-----
-CREATE RESOURCE GROUP my_group WITH (bucket_count = 10) -- normalized!
-CREATE RESOURCE GROUP my_group WITH (bucket_count = (10)) -- fully parenthesized
-CREATE RESOURCE GROUP my_group WITH (bucket_count = _) -- literals removed
-CREATE RESOURCE GROUP _ WITH (bucket_count = 10) -- identifiers removed
-
-parse
 ALTER RESOURCE GROUP my_group WITH (cpu_weight = 200)
 ----
 ALTER RESOURCE GROUP my_group WITH (cpu_weight = 200)
-ALTER RESOURCE GROUP my_group WITH (cpu_weight = (200)) -- fully parenthesized
-ALTER RESOURCE GROUP my_group WITH (cpu_weight = _) -- literals removed
-ALTER RESOURCE GROUP _ WITH (cpu_weight = 200) -- identifiers removed
-
-# No-parens form is also accepted; the canonical form adds them.
-parse
-ALTER RESOURCE GROUP my_group WITH cpu_weight = 200
-----
-ALTER RESOURCE GROUP my_group WITH (cpu_weight = 200) -- normalized!
 ALTER RESOURCE GROUP my_group WITH (cpu_weight = (200)) -- fully parenthesized
 ALTER RESOURCE GROUP my_group WITH (cpu_weight = _) -- literals removed
 ALTER RESOURCE GROUP _ WITH (cpu_weight = 200) -- identifiers removed
@@ -74,14 +47,24 @@ ALTER RESOURCE GROUP IF EXISTS my_group WITH (max_cpu = (true)) -- fully parenth
 ALTER RESOURCE GROUP IF EXISTS my_group WITH (max_cpu = _) -- literals removed
 ALTER RESOURCE GROUP IF EXISTS _ WITH (max_cpu = true) -- identifiers removed
 
-# Ensure BUCKET_COUNT (which triggers WITH_LA in the lexer) works correctly.
-parse
-ALTER RESOURCE GROUP my_group WITH bucket_count = 10
+# The no-parens form is rejected; parentheses are required.
+error
+CREATE RESOURCE GROUP my_group WITH cpu_weight = 100
 ----
-ALTER RESOURCE GROUP my_group WITH (bucket_count = 10) -- normalized!
-ALTER RESOURCE GROUP my_group WITH (bucket_count = (10)) -- fully parenthesized
-ALTER RESOURCE GROUP my_group WITH (bucket_count = _) -- literals removed
-ALTER RESOURCE GROUP _ WITH (bucket_count = 10) -- identifiers removed
+at or near "cpu_weight": syntax error
+DETAIL: source SQL:
+CREATE RESOURCE GROUP my_group WITH cpu_weight = 100
+                                    ^
+HINT: try \h CREATE RESOURCE GROUP
+
+error
+ALTER RESOURCE GROUP my_group WITH cpu_weight = 200
+----
+at or near "cpu_weight": syntax error
+DETAIL: source SQL:
+ALTER RESOURCE GROUP my_group WITH cpu_weight = 200
+                                   ^
+HINT: try \h ALTER RESOURCE GROUP
 
 parse
 DROP RESOURCE GROUP my_group

